crates/polars-expr/src/planner.rs

```diff
@@ -473,10 +473,13 @@ fn create_physical_expr_inner(
             options,
         } => {
             let is_scalar = is_scalar_ae(expression, expr_arena);
-            let output_dtype =
+            let mut output_field =
                 expr_arena
                     .get(expression)
                     .to_field(schema, Context::Default, expr_arena)?;
+            if let Context::Aggregation = ctxt {
```

Comment:
I think we should just pass the context to line 497 instead of updating the dtype here.